src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -257,27 +307,34 @@ internal class ModelCheckingStrategy(
             }
 
         fun isSwitchPosition() = executionPosition in switchPositions
+        fun isCrashPosition() = executionPosition in crashPositions
 
         /**
-         * Creates a new execution position that corresponds to the current switch point.
+         * Creates a new execution position that corresponds to the current switch/crash point.
          * Unlike switch points, the execution position is just a gradually increasing counter
          * which helps to distinguish different switch points.
          */
-        fun newExecutionPosition(iThread: Int) {
+        private fun newExecutionPosition(iThread: Int, isSwitch: Boolean) {
             executionPosition++
-            if (executionPosition > switchPositions.lastOrNull() ?: -1) {
+            if (exploreCrashNode == isSwitch) return
```

Comment:
Lets add `enum ExplorationNodeType` and use it, otherwise this boolean logic seems sophisticated